### ISSUE TITLE
Blacklisting of URLs possible? 

### ISSUE THREAD

**Issue by jo-me** [mammuth/ha-fritzbox-tools#131]
Hey,
do you think it is possible to edit the Fritzbox's URL blacklist?

Reason of my question is:
I want to conveniently block and unblock a certain URL (e.g. only allow for updates).
One way of doing it would be to add and remove it from the blacklist. 
However, without switches in Home assistant the "convenient" part is not given.
 
Pihole or something similar would of course be another option.

**Comment by mammuth** [comment]
I don't know it without testing.
I can recommend the Hassio AdGuard addon. It's quite convenient.

**Comment by SVNKoch** [comment]
I have a similar usecase.
I thought I could do this, then unfortunately found out that the project does not support this (yet).
I was looking for switching the access profile ("Zugangsprofil") of a device.
But changing the blacklist would work too I guess.

Hope this somehow finds it's way in the integration. Thanks for you work!

**Comment by AaronDavidSchneider** [comment]
Thanks for your feedback! I think this will not be possible with our upstream library (https://github.com/AaronDavidSchneider/fritzprofiles).
If you have an idea on how this could work, please feel free to get back to me.
I personally use a pihole. But AdGuard should also work.
Both options are in any way much more convenient than the blacklist tool of the fritzbox!

**Comment by SVNKoch** [comment]
indeed possible!
I did solve my problem using Adguard. Using the adguard home integration.
I set up client (Settings > Client) called "me" and added all devices used by me (by addressing their IP/Mac).
Then I created a file witch contains the urls to block lets say github.com
`||github.com`
but as we don't want this domain blocked on all devices but only for the client we set up earlier I appended the client filter
`||github.com^$client='me'`.
Now we need to add this filter to AdGuard. In theory it cann add these files by URL or any local file. But as far as I've seen the local files don't work at the moment. Thus a url is needed. I used private gists in github to host the file. This way it can only be seen by anyone who has the url.

Adguard exposes certain services in HA to interact with it:
- adguard.enable_url
- adguard.disable_url
- adguard.add_url
- adguard.remove_url
- (adguard.refresh)

We would want to call enable_url and disable_url (same as ticking the checkbox in adguard next to the filter), however these two services are broken currently as adguard now needs the name of the filter in addition to the url.
However we can work arount this using add_url and remove_url.

I use [Node-Red](https://github.com/hassio-addons/addon-node-red) for my automation and call
`adguard.add_url
with 
{"url":"Link to github gist containing blocklist","name":"name of filter"}`
and
`adguard.remove_url
with 
{"url":"Link to github gist containing blocklist"}`

(Note that the service call `adguard.add_url` will result in an error but will work nonetheless)

(using Version 3.0.0 of the AdGuard Home integration)
